ignite/contrib/handlers/tqdm_logger.py

```diff
@@ -199,6 +199,11 @@ def _reset(self, pbar_total):
 
     def _close(self, engine):
         if self.pbar is not None:
+            # https://github.com/tqdm/notebook.py#L240-L250
+            # issue #1115 : notebook backend of tqdm checks if n < total (error or KeyboardInterrupt)
+            # and the bar persists in 'danger' mode
+            if not self.pbar.leave:
```

Comment:
Why we check `if not self.pbar.leave` ?
In the code of `tqdm close` it remains in "danger" state if `self.n < self.total`:
```python
        if self.total and self.n < self.total:
            self.sp(bar_style='danger')
```
for any value of `self.pbar.leave`.

Reply:
Because if the pbar supposed to persist, we should not hack the values. I think.

Reply:
I think even if it should persist it would be better to keep it in "success" state instead of "danger" ?
If it is our call to close the pbar we can say the we update the state to the final step and then close it...

Reply:
If I understand the meaning, we must force the last step of the filtered event ? (I.e. every x batch iterations AND the last batch iteration).

Reply:
yes, in some sense it can be a solution. But I think if our closing event is triggered and we execute our code to close pbar we can simply advance tqdm's `n` to `total` without calling additional pbar updates on the last step.

Reply:
I didn’t catch the difference with what I did. Could you explain a bit more please ? Thanks 😊

Reply:
If you run the following
```python
pbar = ProgressBar(persist=True)

import time
from ignite.engine import Engine, Events

e = Engine(lambda e, b: time.sleep(0.01))

pbar.attach(e, event_name=Events.ITERATION_COMPLETED(every=13))

e.run(range(40), max_epochs=3)
```
you will still have red bars in "danger" state instead of completed bars.

And if you replace the code as 
```python
    def _close(self, engine):
        if self.pbar is not None:
            # https://github.com/tqdm/notebook.py#L240-L250
            # issue #1115 : notebook backend of tqdm checks if n < total (error or KeyboardInterrupt)
            # and the bar persists in 'danger' mode
            if self.pbar.n < self.pbar.total:
                self.pbar.n = self.pbar.total
            self.pbar.close()
        self.pbar = None
```
Bars are closed = green and remain in completed state.

Reply:
Ok I see now, thank you 😊